src/App.js

```diff
@@ -2,18 +2,153 @@ import React from 'react';
 import './App.css';
 
 import users from './api/users';
+import todos from './api/todos';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Static list of todos</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+import TodoList from './components/TodoList/TodoList';
+
+const preparedTodos = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId),
+}));
+
+class App extends React.Component {
+  state = {
+    todos: [...todos],
+    preparedTodos: [...preparedTodos],
+    title: '',
+    userId: null,
+    users: [...users],
+    notChoseUser: false,
+    notEnterTitle: '',
+  }
+
+  handleSubmit = (event) => {
+    if (this.state.userId < 1) {
+      this.setState({
+        notChoseUser: true,
+      });
+    } else if (this.state.notEnterTitle.length < 1) {
+      this.setState({
+        notEnterTitle: true,
+      });
+    } else {
+      this.setState(prevState => ({
+        todos: [
+          ...prevState.todos,
+          {
+            id: prevState.todos.length + 1,
+            userId: Number(prevState.userId),
+            title: prevState.title,
+          },
+        ],
+      }));
+
+      this.setState(prevstate => ({
+        preparedTodos: prevstate.todos.map(todo => ({
+          ...todo,
+          user: users.find(user => user.id === todo.userId),
+        })),
+      }));
+
+      this.setState({
+        title: '',
+        userId: 0,
+      });
+    }
+
+    event.preventDefault();
+  }
+
+  handleChange = (event) => {
```

Comment:
```suggestion
  handleChange = ({ target }) => {
```
если `event` нам не нужен